### DISCUSSION TITLE
Iterative Calculations for Circular References

### DISCUSSION THREAD

**Discussion by rudiseelig** [handsontable/hyperformula#1110] [Feature requests]
Does the library implement iterative calculations to resolve circular references?

Is there an option to set it similar to how Google Sheets / Excel is handling those cases?
<img width="715" alt="Screenshot 2022-11-24 at 21 11 44" src="https://user-images.githubusercontent.com/9105201/203858084-d8e279b1-aaaf-43d5-b1b5-f3b812331bbe.png">

**Comment by sequba** [comment]
HyperFormula does not support iterative calculations at the present development stage. Currently, we prioritize bug and interoperability fixes before any new features, but it is an interesting idea, and we might consider adding it in the future.

Also, bear in mind that HyperFormula is an open-source project. We are open to [contributions from the community](https://hyperformula.handsontable.com/guide/contributing.html), and we will be delighted to collaborate on that feature as it happened recently with https://github.com/handsontable/hyperformula/pull/1047.

**Comment by rudiseelig** [comment]
Thanks for the reply @sequba.

I might fiddle around with this project in the near future to add this feature, but since you are way more familiar with the codebase, could you refer me to a starting point (file/class/etc.) in the code where I could start digging around?

Thanks in advance!

**Reply by sequba** [reply]
@rudiseelig I think you can start by looking at the [Evaluator.recomputeFormulas()](https://github.com/handsontable/hyperformula/blob/25941f6df7c7f2420ac814bacf76f6e31fce94b1/src/Evaluator.ts#L107) method.

This piece of code deals with the cells that belong to some cycle in the [dependency graph](https://hyperformula.handsontable.com/guide/dependency-graph.html#cells-in-the-dependency-graph). Iterative calculation must be run on these cells instead of putting them into the erroneous state:
https://github.com/handsontable/hyperformula/blob/25941f6df7c7f2420ac814bacf76f6e31fce94b1/src/Evaluator.ts#L108-L112

**Comment by mountEvarus** [comment]
I know it's been a while, but I am currently looking into supporting this in my own usage of hyperformula.

Is there still interest in allowing this functionality? (with a flag, of course 😄 )

**Reply by sequba** [reply]
It is a valid feature request. It could be helpful for some of our users, but as far as we know, not many, so currently, the HyperFormula team does not prioritize it.

However, if you are willing to work on such a functionality, we would appreciate your contribution and could offer our help and guidance.

**Reply by mountEvarus** [reply]
Thanks for getting back to quickly, I am currently testing the first version on a fork. Will deploy & make sure it's working first before making the PR.

Will post it here when up 👍

**Reply by mountEvarus** [reply]
I have the PR up now, it's limited in scope. If there is anything I need to change/add, just let me know and I will get to it when I can:

https://github.com/handsontable/hyperformula/pull/1541

**Comment by sequba** [comment]
We are working on this feature together with @mountEvarus 

Issue: https://github.com/handsontable/hyperformula/issues/1545
PR: #1541
